torchdynamo/guards.py

```diff
@@ -238,11 +241,16 @@ def NN_MODULE(self, guard: Guard):
         self.ID_MATCH(guard)
         ref = self.arg_ref(guard)
         val = self.get(guard.name)
-        assert istype(val.training, bool)
-        if val.training:
-            self.code.append(f"{ref}.training")
+
+        def setup_guard():
+            assert istype(val.training, bool)
+            self.code.append(f"___training_state({ref}, {val.training})")
```

Comment:
Is it faster to just directly check the equality here, instead of a function jump to `__training_state` at runtime?

Reply:
I suggested the helper before to allow either hasattr or try/except in the body of it.  Since we aren't doing that I think it is fine to just inline the check.